pillarbox-demo-tv/src/main/java/ch/srgssr/pillarbox/demo/tv/ui/TVDemoTopBar.kt

```diff
@@ -31,21 +37,27 @@ import ch.srgssr.pillarbox.ui.extension.handleDPadKeyEvents
  * Top bar displayed in the demo app on TV.
  *
  * @param destinations The list of destinations to display.
- * @param selectedDestination The currently selected destination.
+ * @param currentNavDestination The currently destination selected.
  * @param modifier The [Modifier] to apply to the top bar.
  * @param onDestinationClick The action to perform the selected a destination.
  */
 @Composable
 @OptIn(ExperimentalComposeUiApi::class, ExperimentalTvMaterial3Api::class)
 fun TVDemoTopBar(
     destinations: List<HomeDestination>,
-    selectedDestination: HomeDestination,
+    currentNavDestination: NavDestination?,
     modifier: Modifier = Modifier,
     onDestinationClick: (destination: HomeDestination) -> Unit
 ) {
     val focusManager = LocalFocusManager.current
-    val focusedTabIndex by rememberSaveable(destinations, selectedDestination) {
-        mutableIntStateOf(destinations.indexOf(selectedDestination))
+    val focusedTabIndex by rememberSaveable(currentNavDestination) {
+        mutableIntStateOf(
+            destinations.indexOfFirst { dest ->
+                currentNavDestination?.hierarchy?.any {
+                    it.route == dest.route
+                } == true
```

Comment:
`currentNavDestination?.hierarchy` generates a new `Sequence` every time it is called. Maybe store it in a variable before `indexOfFirst`?
```suggestion
        val destinationHierarchy = currentNavDestination?.hierarchy.orEmpty()

        mutableIntStateOf(
            destinations.indexOfFirst { dest ->
                destinationHierarchy.any { it.route == dest.route }
```